include/backend/drm/drm.h

```diff
@@ -25,11 +25,17 @@ struct wlr_drm_plane {
 	struct wlr_drm_surface surf;
 	struct wlr_drm_surface mgpu_surf;
 
+	/* Buffer to be commited on the next page flip */
+	struct wlr_drm_fb queued_fb;
+	/* Buffer we have already committed */
+	struct wlr_drm_fb committed_fb;
+	/* Buffer we committed and have been notified about */
+	struct wlr_drm_fb complete_fb;
```

Comment:
Why do we need to keep this around? Shouldn't screen capture use `committed_fb`?

Reply:
Adding this was actually the fix to the tearing issue with scanned out buffers. Releasing `committed_fb` inside of `page_flip_handler` or automatically when `committed_fb` is overwritten on the next commit always lead to rendering artifacts.
I honestly don't entirely understand the mechanics of it.

Reply:
Actually I think the buffer handling is wrong here. There are three states: https://github.com/emersion/glider/blob/c66847dd1cf8ae5e68666ce7cb3be41427b38dc7/include/backend/backend.h#L51

- Pending: will be used on next page-flip
- Queued: has been used on the last page-flip, but hasn't been displayed yet on screen
- Current: is being shown on screen